client/components/test-mode-notice/test/index.js

```diff
@@ -92,89 +91,15 @@ describe( 'Test mode notification', () => {
 		}
 	);
 
-	test.each( allTopics )( 'Returns correct notice', ( topic ) => {
-		const expected = (
-			<Notice
-				className="wcpay-test-mode-notice"
-				status="warning"
-				isDismissible={ false }
-			>
-				{ getNoticeMessage( topic ) }
-			</Notice>
-		);
-
-		expect( getNotice( topic ) ).toStrictEqual( expected );
-	} );
```

Comment:
I think we should keep this test; just set `isInTestMode` to return true via `isInTestMode.mockReturnValue( true )`.

In addition to that we should add a new test verifying that no notice is returned when `isInTestMode` returns `false`. Something along the lines of:

```js
isInTestMode.mockReturnValue( false );
expect( <TestModeNotice topic={ /* choose any topic here */ } /> ).toBeUndefined(); // .toBeNull() maybe?
```